core/chains/evm/gas/models.go

```diff
@@ -328,7 +323,7 @@ func (e *evmFeeEstimator) GetMaxCost(ctx context.Context, amount assets.Eth, cal
 
 func (e *evmFeeEstimator) BumpFee(ctx context.Context, originalFee EvmFee, feeLimit uint64, maxFeePrice *assets.Wei, attempts []EvmPriorAttempt) (bumpedFee EvmFee, chainSpecificFeeLimit uint64, err error) {
 	// validate only 1 fee type is present
-	if (!originalFee.ValidDynamic() && originalFee.Legacy == nil) || (originalFee.ValidDynamic() && originalFee.Legacy != nil) {
+	if (!originalFee.ValidDynamic() && originalFee.GasPrice == nil) || (originalFee.ValidDynamic() && originalFee.GasPrice != nil) {
 		err = pkgerrors.New("only one dynamic or legacy fee can be defined")
```

Comment:
Since we're replacing `legacy` with gas price, does this error need to change?

Reply:
Do you mean the logic or the error message?

Reply:
Just the error message

Reply:
I think the terms legacy and dynamic transactions remain relevant, it's just the underlying representation that is named differently. However, since we mention the word `fee` here, I'll update it to make it more accurate.